private/tools/java/rules/jvm/external/Hasher.java

```diff
@@ -17,20 +17,38 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-/** A tool to compute the sha256 hash of a file. */
+/**
+ * A tool to compute the sha256 hash of a file.
+ */
 public class Hasher {
 
   public static void main(String[] args) throws NoSuchAlgorithmException, IOException {
-    System.out.print(hashFiles(args));
+    Stream<String> fileStream = argsToStream(args);
+    System.out.print(hashFiles(fileStream));
   }
 
-  static String hashFiles(String[] files) {
-    return Stream.of(files)
+  static Stream<String> argsToStream(String[] args) throws IOException {
+    Stream<String> fileStream;
+
```

Comment:
since this hasher is only used for rules_jvm_external, I don't think there's a need to keep the original non-argsfile approach.

Reply:
Removed the non-argsfile approach and had it raise an IAE